### PR TITLE
Fixed Kriging docstrings 

### PR DESCRIPTION
**Pull request by sofianehaddad**
No more collection of {Basis, Point}
No more transformation

### CODE REVIEW

File: python/src/GeneralLinearModelResult_doc.i.in [hunk 1]
```diff
@@ -11,11 +11,11 @@ residuals : :class:`~openturns.Point`
     The residual errors.
 relativeErrors : :class:`~openturns.Point`
     The relative errors.
-basis : collection of :class:`~openturns.Basis`
-    Collection of the  :math:`p` functional basis: :math:`(\varphi_j^l: \Rset^d \rightarrow \Rset)_{1 \leq j \leq n_l}` for each :math:`l \in [1, p]`.
+basis :  :class:`~openturns.Basis`
+    Collection of the  :math:`b` functional basis: :math:`(\varphi^l: \Rset^d \rightarrow \Rset^p)` for each :math:`l \in [1, b]`.
```
Comment (jschueller) [inline python/src/GeneralLinearModelResult_doc.i.in:15]: I think we forgot to mention the api changes in the changelog, could you add it ?

File: python/src/GeneralLinearModelResult_doc.i.in [hunk 2]
```diff
@@ -11,11 +11,11 @@ residuals : :class:`~openturns.Point`
...
     Its size should be equal to zero if the trend is not estimated.
-trendCoefficients : collection of :class:`~openturns.Point`
-   The trend coefficients vectors :math:`(\vect{\alpha}^1, \dots, \vect{\alpha}^p)`.
+trendCoefficients :  :class:`~openturns.Point`
```
[the remark this answers is not in the document]
Comment (sofianehaddad) [reply python/src/GeneralLinearModelResult_doc.i.in:17]: True. However I copy/paste the `residuals` content 
As it imapct the `XXXResult` structure, that is created by `XXXAlgo` class, I think we can keep think like that.
What do you think ?

[the remark this answers is not in the document]
Comment (jschueller) [reply python/src/GeneralLinearModelResult_doc.i.in:17]: ok, up to you

File: ChangeLog [hunk 3]
```diff
@@ -26,6 +26,7 @@
  * Removed Sample.storeToTemporaryFile|streamToRFormat methods
  * Removed GaussianProcess.GIBBS
  * Deprecated DomainIntersection|Union|DisjunctiveUnion(Domain, Domain) ctors
+ * Dropped usage of collection of Point + collection of Basis in GeneralLinearModelAlgorithm | KrigingAlgorithm
```
Comment (jschueller) [inline ChangeLog:29]: * KrigingResult.getTrendCoefficients now returns a single Point
* KrigingResult.getBasisCollection was removed in favor of KrigingResult.getBasis which returns a single Basis

**Comment by JPelamatti** [comment]
If the constructor now only takes a basis object as input, and not a basis collection, shouldn't we also replace _Collection of the  :math:`b` functional basis_ by  _ functional basis of size :math:`b`_?

**Comment by sofianehaddad** [comment]
Thanks all for your feedback. I took into account your remarks & now should be ok

**Review by jschueller — APPROVED** [review]

File: python/src/GeneralLinearModelResult_doc.i.in [hunk 4]
```diff
@@ -109,13 +109,13 @@ covModel : :class:`~openturns.CovarianceModel`
...
 
-%feature("docstring") OT::GeneralLinearModelResult::getBasisCollection
+%feature("docstring") OT::GeneralLinearModelResult::getBasis
 "Accessor to the collection of basis.
 
 Returns
 -------
-basisCollection : collection of :class:`~openturns.Basis`
-    Collection of the :math:`p` function basis: :math:`(\varphi_j^l: \Rset^d \rightarrow \Rset)_{1 \leq j \leq n_l}` for each :math:`l \in [1, p]`.
+basisCollection : :class:`~openturns.Basis`
```
Comment (JPelamatti) [inline python/src/GeneralLinearModelResult_doc.i.in:117]: I think the type here is is wrong

File: python/src/GeneralLinearModelResult_doc.i.in [hunk 5]
```diff
@@ -109,13 +109,13 @@ covModel : :class:`~openturns.CovarianceModel`
 
 // ---------------------------------------------------------------------
 
-%feature("docstring") OT::GeneralLinearModelResult::getBasisCollection
+%feature("docstring") OT::GeneralLinearModelResult::getBasis
 "Accessor to the collection of basis.
```
Comment (JPelamatti) [inline python/src/GeneralLinearModelResult_doc.i.in:113]: accessor to the functional basis?

File: python/src/KrigingResult_doc.i.in [hunk 6]
```diff
@@ -132,13 +130,13 @@ covModel : :class:`~openturns.CovarianceModel`
 
 // ---------------------------------------------------------------------
 
-%feature("docstring") OT::KrigingResult::getBasisCollection
+%feature("docstring") OT::KrigingResult::getBasis
 "Accessor to the collection of basis.
```
Comment (JPelamatti) [inline python/src/KrigingResult_doc.i.in:134]: accessor to the functional basis?

Comment (sofianehaddad) [reply python/src/KrigingResult_doc.i.in:134]: indeed

File: python/src/GeneralLinearModelResult_doc.i.in (hunk 4 above)
Comment (sofianehaddad) [reply python/src/GeneralLinearModelResult_doc.i.in:117]: why ? It is a basis

**Review by JPelamatti — APPROVED** [review]

File: python/src/GeneralLinearModelResult_doc.i.in (hunk 4 above)
Comment (JPelamatti) [reply python/src/GeneralLinearModelResult_doc.i.in]: In the same way as for the krigingresult, shoudln't it be :  basis : :class:`~openturns.Basis` ?

File: lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingResult.cxx [hunk 7]
```diff
@@ -181,7 +177,7 @@ void KrigingResult::computeF() const
   // Nothing to do if the design matrix has already been computed
   if (F_.getNbRows() != 0) return;
   const UnsignedInteger outputDimension = covarianceModel_.getOutputDimension();
-  const UnsignedInteger sampleSize = inputSample_.getSize();
+  const UnsignedInteger sampleSize = getInputSample().getSize();
```
Comment (JPelamatti) [inline lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingResult.cxx:180]: The  MetaModelResult parent class (at least through the constructor that is used in this case) has an inputSample_ attribute, isn't it easier to keep the code here as it is?

Comment (sofianehaddad) [reply lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingResult.cxx:180]: yes it has but the attribut is private. Thus no choice....

Comment (JPelamatti) [reply lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingResult.cxx:180]: oh, I see

**Review by JPelamatti — APPROVED** [review]

**Review by regislebrun — APPROVED** [review]

**Pull request merged**